### DISCUSSION TITLE
Wrong consumption with external generation

### DISCUSSION THREAD

**Discussion by DaSpors** [weltenwort/home-assistant-rct-power-integration#324] [General]
If have a problem with two hoymiles inverters.
Measurements are reported to the RCT from a power meter (DRT428DC-V3) using a S0 connection.

In the RCT App it shows everything corretly, but in my HomeAssistant the consumption gets negative.

So while I'm able to get the current power from the hoymiles inverters (from an OpenDTU) and the production is reported via S0 correctly to the RCT inverter, I'm unable to calculate correct consumtion values in HA.

![grafik](https://github.com/weltenwort/home-assistant-rct-power-integration/assets/623201/6e27a4ca-c5b0-4cda-a0f5-83bfa1c0c569)

![grafik](https://github.com/weltenwort/home-assistant-rct-power-integration/assets/623201/91316b75-b792-462d-bf06-b823bb90c46c)

Any chance to get the consumption values as shown in the RTCT App?

**Comment by heikone** [comment]
Hallo Daniel, 
leider ein wenig Offtopic zu Deiner Anfrage, aber:
Du scheinst da etwas umgesetzt zu haben, was ich auch plane. Nämlich ein oder zwei externe Wechselrichter als Erzeuger an das RCT-System zu bringen.
Eine Rückfrage dazu bei RCT verwies mich auf die Anbindung mit Modbus, aber die Anbindung mit S0 scheint mir einfacher.
Kannst Du Deun Setup beschreiben? Das wäre für mich eine große Hilfe, wie ich es umsetzen könnte. 
Gruß,
Heiko

**Reply by DaSpors** [reply]
Hier ist mein Setup als Schema-Zeichung:
![PV Ist _ Planung](https://github.com/weltenwort/home-assistant-rct-power-integration/assets/623201/7e538c50-b595-4830-8315-4e1002b17df0)

Interessant eigentlich nur der orangene Kram und der RCT Power Inverter.
Z3 ist der DRT428DC-V3, dessen S0 am Inverter angeschlossen ist.
Dazu im Handbuch vom RCT-Power-Storage auf Seite 90 das Verkabelungsschema und auf Seite 91 die dazugehörigen Einstellungen (in der App).
https://www.rct-power.com/de/download-bereich-de.html?file=files/Download-Bereich/Download%20Bereich%20DE/3_RCT%20Power%20Storage%20DC%204-6/RCT-Power-Storage-DC_Manual_Web21V2DE.pdf

**Comment by heikone** [comment]
Dankeschön - das wäre schon mal eine Vorlage für mich! 
Ich habe zwar einen RCT Power Storage 8.0 DC im Einsatz, aber da scheint der Anschluss über S0 auf die gleiche Art und Weise zu sein.
Ich bin nur am überlegen, ob nicht ein Zähler mit Modbus Vorteile hätte.
Mein Problem ist, dass ich nicht so ohne weiteres ein Kabel von der Fremd-Solar-Anlage zum RCT-Wechselrichter gelegt bekomme. Für Modbus gibt es zumindest Funkbrücken ... aber vermutlich auch nicht leichter umzusetzen ein wenig Fummelei zum verlegen des Kabels. Ob nun Modbus oder S0-Leitung ist dann auch egal.
Aber gut zu wissen, wie es schon mal mit S0 gehen würde.
Gruß, Heiko
